RecoPixelVertexing/PixelTrackFitting/src/PixelTrackFilterByKinematics.cc

```diff
@@ -27,8 +27,8 @@ bool PixelTrackFilterByKinematics::operator()(const reco::Track* ptrack, const P
 {
   if (!ptrack) return false;
   auto const & track = *ptrack;
-  if (track.chi2() > theChi2Max) return false;
```

Comment:
I am curious, why the swap ?

Reply:
It works better, because the cut on Tip is larger than the cut on chi2. Swapping them, the cut on chi2 becomes nearly useless after the cut on Tip, because large Tip also means large chi2.

Cutting on chi2 first would instead leave a lot more fakes, to be removed by the cut on Tip.

Reply:
Is it true in general (i.e. also for the Riemann fit) or only for the broken line fit ?

Reply:
I have tried it only for the broken line fit, but I strongly expect that it's true in general. In fact, in other words, the reason for the swap is that I noticed that the correlation between fake rate and Tip is greater than the correlation between fake rate and Chi2. Although the latter certainly varies between different fits (and so the cut on Chi2 must be set differently), I find hard to imagine that a fake track with low chi2 according to broken line has instead a very high chi2 according to another fit.

Reply:
Unless I'm missing something, physics-wise the result is the same (= "and of the two cuts"). Technically `track,chi2()` only returns a member variable, whereas the Tip cut contains also some math, so the original order (chi2, tip) should be tiny bit faster than (tip, chi2).

Reply:
Oh, my bad, you are right. The improvement is on *tuning* the cut on chi2 after tuning the one on tip. I swapped them in the code when I noticed that tuning the tip first gave better results, but indeed the swap in the code has no effect.